checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallInference.java

```diff
@@ -241,6 +300,17 @@ private void addMemberAndClassAnnotations() {
       addEnsuresCalledMethods();
     }
 
+    // If all return statements alias the same parameter index, then add the @MustCallAlias
+    // annotation to that parameter.
+    if (!returnNodeToParameterIndexMap.isEmpty()) {
+      if (returnNodeToParameterIndexMap.values().stream().distinct().count() == 1) {
+        int indexOfParam = returnNodeToParameterIndexMap.values().iterator().next();
+        if (indexOfParam > 0) {
```

Comment:
Why not permit inference of `@MustCallAlias` on the receiver? I think that could be a sensible specification.

If there is a technical reason (I suspect there probably is!), please add a comment explaining it.

Reply:
We didn't have a rule for the receiver type and I didn't consider this case, but I added some changes to support it. Thanks for bringing this to my attention.